lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -202,8 +206,8 @@ defmodule Mix.Tasks.Compile.App do
         :ok
       {:mod, value} ->
         Mix.raise "Application callback module (:mod) should be either [] or {module, start_args}, got: #{inspect value}"
-      _ ->
-        :ok
+      {opt, _value} when is_atom(opt) ->
```

Comment:
I don't think we should raise because it is totally legal to put any key you want in the application file.